File: mock-facility-management/src/main/java/de/vkb/dojo/es/facilityManagement/services/events/EventAggregator.java
package de.vkb.dojo.es.facilityManagement.services.events;


import de.vkb.dojo.es.facilityManagement.model.event.Event;

public interface EventAggregator<E extends Event, A> {
    EventAggregatorResult<E, A> process(E event, A aggregate);
}

File: mock-human-resources/src/main/kotlin/de/vkb/dojo/es/humanResources/services/events/EventAggregator.kt
package de.vkb.dojo.es.humanResources.services.events

import de.vkb.dojo.es.humanResources.model.event.Event

interface EventAggregator<E: Event, A> {
    fun process(event: E, aggregate: A?): EventAggregatorResult<E, A>
}
